src/train_model.py

```diff
@@ -126,6 +127,51 @@ def train_model(config: Struct):
     print(f"-log(1 / {config.vocab_size}) = " + \
         f"{-torch.log(torch.tensor(1 / config.vocab_size))}")
 
+    # Checks to see if you want to use a portion of the dataset for training
+    if config.split_dataset:
+        print(f"\nUsing {config.split_dataset_percentage}% of the dataset for training")
+
+        # Define the source and target directories
+        source_dir = Path("/grphome/grp_retnet/compute/data/datasets_tokenized/c4/train")
```

Comment:
This is hardcoded. Please make this work without hardcoding the file path to the source and target directories.